components/controls/RadioToggle.tsx

```diff
@@ -18,52 +18,58 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 import * as classNames from 'classnames';
+import * as _ from 'lodash';
 import * as React from 'react';
 import './RadioToggle.css';
 import Tooltip from './Tooltip';
 
-interface Option {
+interface Option<T = string> {
   disabled?: boolean;
   label: string;
   tooltip?: string;
-  value: string;
+  value: T;
 }
 
-interface Props {
+interface Props<T = string> {
   className?: string;
   name: string;
-  onCheck: (value: string) => void;
-  options: Option[];
-  value?: string;
+  onCheck: (value: T | undefined) => void;
+  options: Option<T>[];
+  value?: T;
 }
 
-export default class RadioToggle extends React.PureComponent<Props> {
+export default class RadioToggle<T = string> extends React.PureComponent<Props<T>> {
   static defaultProps = {
     disabled: false,
     value: null
   };
 
   handleChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
     const newValue = e.currentTarget.value;
-    this.props.onCheck(newValue);
+    // Let assum that if we can't found the option we have unselect
+    // RadioToggle
+    const selectedOption = _.find(this.props.options, { label: newValue });
+    this.props.onCheck(selectedOption && selectedOption.value);
   };
 
-  renderOption = (option: Option) => {
+  renderOption = (option: Option<T>) => {
     const checked = option.value === this.props.value;
     const htmlId = this.props.name + '__' + option.value;
     return (
-      <li key={option.value}>
+      <li key={option.label}>
```

Comment:
I'm not sure we should really use the label for this and the value... I guess it's only a coincidence that the label is a `string` here, it could be as well be a `ReactNode` to let some flexibility on what we can display...

In fact, reading the ticket, I'm not sure to understand why we want to be able to handle anything, is there a use case behind it @jeremy-davis-sonarsource ? Maybe it should be limited to string, number or an object that would have a key. So that we can use the key of that object as the value.

Reply:
I'm also wondering why we need to improve the typing of this value. I knew it but now I can't remember. Can you enlighten us ?

Reply:
As we have discuss we can use controllled componant.